macros/base/park.cfg

```diff
@@ -27,21 +26,35 @@ gcode:
     _CG28 ; home if not already homed
 
     {% if printer.extruder.can_extrude %}
-        {% if firmware_retraction_enabled %}    # use firmware_retraction parameter for retract (in case firmware retraction is selected in printer.cfg)
-            {% if verbose %}
-                RESPOND MSG="Firmware retraction enabled, Extruder retraction = {printer.firmware_retraction.retract_length}"
-            {% endif %}
-            G10
-        {% else %}                              # otherwise:
-            {% if MATERIAL != "XXX" %}          # use material parameter if available for retract, otherwise use default value
-                {% set material = printer["gcode_macro _USER_VARIABLES"].material_parameters[MATERIAL] %}
-                {% set E = material.retract_length %}
-            {% endif %}
+        {% if params.E is defined %}
+            {% set E = params.E|float|abs %}
+
             {% if verbose %}
-                RESPOND MSG="Firmware retraction disabled, Extruder retraction = {E}"
+                RESPOND MSG="Retraction overrided with parameter, Extruder retraction = {E}"
+            {% endif %}            
+        {% else %}
+            {% if firmware_retraction_enabled %}    # use firmware_retraction parameter for retract (in case firmware retraction is selected in printer.cfg)
+                {% if verbose %}
+                    RESPOND MSG="Firmware retraction enabled, Extruder retraction = {printer.firmware_retraction.retract_length}"
+                {% endif %}
+                G10
+            {% else %}                              # otherwise:
+                {% if printer["gcode_macro _USER_VARIABLES"].material_parameters[material] is defined %}          # use material parameter if available for retract, otherwise use default value
+                    {% set E = printer["gcode_macro _USER_VARIABLES"].material_parameters[material].retract_length|default(1.7) %}
+                {% else %}
+                    {% set E = 1.7 %}
+                {% endif %}
+
+                {% if verbose %}
+                    RESPOND MSG="Firmware retraction disabled, Extruder retraction = {E}"
+                {% endif %}
             {% endif %}
+        {% endif %}
+
+        {% if E is defined and E > 0 %}
             G92 E0
             G1 E-{E} F2100
+            RESPOND MSG="Extruder retraction = {E}"
```

Comment:
this one

Reply:
This respond is useless, so I've deleted it. It was introduced to test my code.